src/services/dataParser.ts

```
@@ -0,0 +1,106 @@
+  private static parsers = new Map<string, DataParser<unknown, unknown>>();
+
+  /**
+   * Register a data parser for a specific tile type
+   * @param tileType - Unique identifier for the tile type
+   * @param parser - Data parser instance
+   */
+  static register<TTileType extends string, TRawData, TTileData>(
+    tileType: TTileType,
+    parser: DataParser<TRawData, TTileData>,
+  ): void {
+    this.parsers.set(tileType, parser as DataParser<unknown, unknown>);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)